design/src/genre-analysis.md

```diff
@@ -0,0 +1,121 @@
+# Genre Analysis: Factory Builders
+
+## Defining Games
+
+- Factorio
+  - colonize an alien planet and launch a rocket
+- Satisfactory
+  - open world first person factory building
+- Shapez.io
+  - abstract factory puzzles
+- Dyson Sphere Program
+  - gather resources from an entire solar system
+
+## Defining characteristics
+
+- automated resource extraction
+- automation resource refinement
+- micro-optimizations
+- gated technology progression
+
+## Adjacent Genres
+
+- Base and city builders
+  - Build out a complex and delightful area, but with no / little automation
+- Zachlike puzzles
+  - Micro-optimization as an entire game
+- Colony sims
+  - Balance complex systems, but with a focus on the needs of your workers, not scaling production
+- Idle games
+  - Automate everything, but without the complex chains
+- Train simulators
+  - Focusing almost entirely on logistic networks powered by trains
+
+## Blended Genre Games
+
+- Mindustry (tower defense)
+- Hydroneer | Astroneeer | Atrio (exploration)
+- Automachef (Zachlike + cooking game)
+- Infinifactory (Zachlike)
+- Timberborn (city builder)
+
+## Progression Loop
+
+- explore
+- find resource patch
+- exploit resource
+- refine resource
+- build into science
+- unlock new technology
+- optimize base with new technology
+- explore to find new resource type to exploit
+
+## Sources of Challenge
+
+- player-driven optimization
+- resource refinement puzzles
+- logistics layout
+- (optional) combat disrupting operations
+- (optional) temporal variation creating unstable
+
+## Design Tensions
+
+- complexity vs meaningful richness
+- helpful innovations vs trivializing the game
+- disruption forces interesting changes vs frustration and sadness to see your stuff get wrecked
+- long play time vs tedious waiting around
+- player fantasy of being in the game vs annoyance of inventory space and walking times
+- giant automated bases vs computational limits
```

Comment:
On this point: do you know what kind of FPS the games listed in Defining Games end up managing in the long-term?

Reply:
People play them all the way down to like 1 FPS. Much much harder if there are realtime elements to respond to though.

And you need good frame pacing.

Reply:
Notably, in Factorio players always talk about two different concepts: Frames per second (FPS) and updates per second (UPS).
Then for large factories they try to optimize the UPS.

According to this random forum comment, the RAM latency seems to be a bottleneck for Factorio, which I didn't expect.
<https://forums.factorio.com/viewtopic.php?p=226062#p226062>